cloud/ec2.go

```diff
@@ -1197,7 +1197,7 @@ func (m *ec2Manager) StopInstance(ctx context.Context, h *host.Host, user string
 		if currentStatusErr != nil {
 			return errors.Wrap(currentStatusErr, "error getting instance status after stopping error")
 		}
-		if currentStatusErr := h.SetStatus(currentStatus.String(), user, "reverting to the most up-to-date host status"); currentStatusErr != nil {
+		if currentStatusErr := h.SetStatusAtomically(currentStatus.String(), user, "reverting to the most up-to-date host status"); currentStatusErr != nil {
```

Comment:
I don't fully understand how this change makes it only set the status if the host status has not been changed since it tried to stop the host. Where does that checking take place?

Reply:
`SetStatusAtomically` will only update the host status if the host status in the DB is still "stopping". If it's another state, the host's state has been changed by something other than this thread and attempting to overwrite that state change would be undesirable.

Reply:
Ah I see that now. Thank you for explaining it.